<a href="https://colab.research.google.com/github/jeanbaptistehabimanaMIT2026/MIT-projects-2026/blob/main/Jean%20%20Baptiste%20%20HABIMANA%2C%20%20%20%20%20RegNo%3A%20M05688/2026%2C%20%20%20Practical%20Works%20of%20Data%20Mining%20for%20Business%20Intelligence.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

  UNIVERSITY OF LAY ADVENTISTS OF KIGALI
  NAME: Jean Baptiste HABIMANA
  REGNO: M05688/2026

  MODULE: C11712(DATA MINING FOR BUSINESS INTELLIGENCE)
  
  DATE: 28/06/2026


 **PRACTICAL WORKS OF DATA MINING FOR BUSINESS INTELLIGENCE**


** MY TURN: NumPy Practice**

1. Create the unit_prices array

In [ ]:
import numpy as np

In [ ]:

unit_prices = np.array([450, 1200, 3000, 750, 2100, 980])
print("Unit prices:", unit_prices)


Unit prices: [ 450 1200 3000  750 2100  980]


 2. Print the total (sum)

In [ ]:
print("Total sum:", np.sum(unit_prices))

Total sum: 8480


 3. Print the average (mean), rounded

In [ ]:
print("Average price (rounded):", round(np.mean(unit_prices)))

Average price (rounded): 1413


4. Apply 5% price increase

In [ ]:
increased_prices = unit_prices * 1.05
print("Increased prices:", increased_prices)


Increased prices: [ 472.5 1260.  3150.   787.5 2205.  1029. ]


 5. Boolean mask for prices > 1000

In [ ]:
mask = unit_prices > 1000
print("Prices greater than 1000:", unit_prices[mask])


Prices greater than 1000: [1200 3000 2100]


         OUTPUT:
         unit prices: [450  1200  3000  750  2100  980]
         Total sum: 8480
         Average price (rounded): 1413
         Increased prices: [ 472.5  1260.  3150. 787.5  2205.  1029. ]
         Prices greater than 1000: [1200  3000  2100]


# Comments on NumPy practice on data given for Business Intelligence context for each step:

1. Array creation: loads raw product pricing data into a structure NumPy can analyze at scale

2. Sum :  total revenue potential if one unit of each item is sold ($8,480)

3. Mean:  the average price point (~$1,413) helps benchmark new products or flag outliers

4. 5% increase:  simulates a price adjustment scenario; vectorized operations make this instant across thousands of SKUs

5. Boolean mask : filters the "premium" segment (>$1,000), useful for targeted marketing or tiered discount strategies

              Thanks.


**MY TURN: Filtering & Sorting Practice**

6. Filter df to show only rows where category is 'Produce'

In [7]:
import pandas as pd

In [8]:
df=pd.read_csv("supermarket_sales.csv")

In [9]:
df.head()

,Date,Product,Category,Region,UnitPrice,QuantitySold
0,2025-01-01,Sugar,Grocery,Musanze,2113.0,41.0
1,2025-01-01,Soda,Beverages,Musanze,532.0,25.0
2,2025-01-02,Rice,Grains,Rubavu,2481.0,7.0
3,2025-01-02,Bread,Bakery,Rubavu,4844.0,5.0
4,2025-01-02,Soda,Beverages,Kigali,1830.0,6.0


In [14]:
df[df['Category'] == 'Produce']

,Date,Product,Category,Region,UnitPrice,QuantitySold
5,2025-01-03,Tomatoes,Produce,Musanze,4492.0,1.0
8,2025-01-04,Tomatoes,Produce,Rubavu,3429.0,2.0
20,2025-01-07,Onions,Produce,Rubavu,2523.0,33.0
24,2025-01-07,Onions,Produce,Rubavu,2277.0,8.0
32,2025-01-11,Tomatoes,Produce,Huye,1590.0,38.0
35,2025-01-12,Tomatoes,Produce,Huye,1280.0,NaN
39,2025-01-13,Onions,Produce,Kigali,4110.0,1.0
43,2025-01-16,Tomatoes,Produce,Musanze,1382.0,40.0
52,2025-01-18,Tomatoes,Produce,Kigali,3541.0,25.0


In [15]:
print(df.columns)

Index(['Date', 'Product', 'Category', 'Region', 'UnitPrice', 'QuantitySold'], dtype='object')


7. Filter df to show only rows where QuantitySold is greater than or equal to 20 AND  Region is 'Musanze

In [16]:
df[(df['QuantitySold'] >= 20) & (df['Region'] == 'Musanze')]

,Date,Product,Category,Region,UnitPrice,QuantitySold
0,2025-01-01,Sugar,Grocery,Musanze,2113.0,41.0
1,2025-01-01,Soda,Beverages,Musanze,532.0,25.0
34,2025-01-12,Rice,Grains,Musanze,4475.0,28.0
38,2025-01-12,Cooking Oil,Grocery,Musanze,NaN,21.0
43,2025-01-16,Tomatoes,Produce,Musanze,1382.0,40.0
49,2025-01-17,Bread,Bakery,Musanze,2762.0,23.0


8. Create a new column called RevenuePerUnit equal to UnitPrice

In [17]:
df['RevenuePerUnit'] = df['UnitPrice']

9. Sort the DataFrame by Date (oldest first) and show the first 10 rows

In [18]:
df.sort_values(by='Date').head(10)

,Date,Product,Category,Region,UnitPrice,QuantitySold,RevenuePerUnit
0,2025-01-01,Sugar,Grocery,Musanze,2113.0,41.0,2113.0
1,2025-01-01,Soda,Beverages,Musanze,532.0,25.0,532.0
2,2025-01-02,Rice,Grains,Rubavu,2481.0,7.0,2481.0
3,2025-01-02,Bread,Bakery,Rubavu,4844.0,5.0,4844.0
4,2025-01-02,Soda,Beverages,Kigali,1830.0,6.0,1830.0
5,2025-01-03,Tomatoes,Produce,Musanze,4492.0,1.0,4492.0
6,2025-01-03,Bread,Bakery,Rubavu,3057.0,12.0,3057.0
7,2025-01-03,Sugar,Grocery,Huye,3710.0,1.0,3710.0
8,2025-01-04,Tomatoes,Produce,Rubavu,3429.0,2.0,3429.0
9,2025-01-04,Soap,Household,Huye,3627.0,28.0,3627.0


10. Find all rows where Product is either 'Milk', 'Eggs', or 'Soda' using isin()

In [19]:
df[df['Product'].isin(['Milk' , 'Eggs' , 'Soda'])]

,Date,Product,Category,Region,UnitPrice,QuantitySold,RevenuePerUnit
1,2025-01-01,Soda,Beverages,Musanze,532.0,25.0,532.0
4,2025-01-02,Soda,Beverages,Kigali,1830.0,6.0,1830.0
10,2025-01-04,Milk,Dairy,Rubavu,3899.0,6.0,3899.0
12,2025-01-05,Eggs,Dairy,Huye,1267.0,39.0,1267.0
14,2025-01-05,Soda,Beverages,Huye,3969.0,39.0,3969.0
19,2025-01-07,Soda,Beverages,Huye,950.0,11.0,950.0
21,2025-01-07,Soda,Beverages,Rubavu,2998.0,49.0,2998.0
25,2025-01-07,Milk,Dairy,Huye,1854.0,49.0,1854.0
28,2025-01-09,Soda,Beverages,NaN,2477.0,32.0,2477.0
30,2025-01-09,Eggs,Dairy,Huye,1241.0,27.0,1241.0


These are the standard Pandas solutions for my turn called Filtering & Sorting practice.


      Thanks.